### DISCUSSION TITLE
How to run julia code in positron and RStudio by qmd? How to use QuartoNotebookRunner in positron and RStudio by qmd? https://github.com/PumasAI/QuartoNotebookRunner.jl

### DISCUSSION THREAD

**Discussion by Liu23308** [quarto-dev/quarto-cli#11742] [Q&A]
### Description

How to run julia code in positron and RStudio by qmd? How to use QuartoNotebookRunner in positron and RStudio by qmd? https://github.com/PumasAI/QuartoNotebookRunner.jl

**Comment by Liu23308** [comment]
use Julia

```{julia}

1+2

```

Error in julia_setup() : Julia is not found.

**Reply by cderv** [reply]
> Error in julia_setup() : Julia is not found.

Judging by this line, it seems you are using a document which triggers the **knitr** engine, and so that will handle any Julia cell with **knitr** and call [`JuliaCall`](https://github.com/JuliaInterop/JuliaCall). 

> How to use QuartoNotebookRunner

As you are asking this question, you may want to use the **julia** engine directly with Quarto. For this, there is not automatic binding of engine yet, and  you need to add `engine: julia`as documented in link provided below. 

The third way to use Julia is through Jupyter with IJulia and so using the `engine: jupyter` in Quarto. 

Hope this clarify how it works to help you see more clearly into the doc. 

Thanks !

**Comment by mcanouil** [comment]
See the documentation: https://quarto.org/docs/computations/julia.html#using-the-julia-engine

@Liu23308 in the future, please consider searching and reading the documentation before asking questions. If you still have issues, please do search first for similar issues in GitHub discussions and GitHub Issues.
Finally, please read the guidelines and links (e.g., <https://quarto.org/bug-reports.html>) provided in the issues/discussions templates when you are opening a new discussion/issue.
